eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php

```diff
@@ -52,14 +47,11 @@ public function downloadBinaryFileAction( $contentId, $fieldIdentifier, $filenam
             $content = $this->contentService->loadContent( $contentId );
         }
 
-        $field = $this->translationHelper->getTranslatedField(
-            $content, $fieldIdentifier,
-            $request->query->has( 'language' ) ? $request->query->get( 'language' ) : null
-        );
-        if ( !$field instanceof Field )
+        $field = $this->findField( $content, $fieldId );
+        if ( $field === false )
         {
             throw new InvalidArgumentException(
-                "'{$fieldIdentifier}' field not present on content #{$content->contentInfo->id} '{$content->contentInfo->name}'"
+                "'No field with id $fieldId found in content #{$content->contentInfo->id} '{$content->contentInfo->name}'"
```

Comment:
I think this belongs to `findField()`. This would allow it to only return a Field object. Returning `false` on error is evil :smiling_imp: